### ISSUE TITLE
Cannot iterate over the contents of multiple matched XPath elements

### ISSUE THREAD

**Issue by hvt** [hakre/XMLReaderIterator#12]
Nice library hakre.

Maybe I'm using it wrong and is it by design that this happens, or maybe I must use another iterator. This is however what I'm trying to achieve.

Suppose this XML document:

``` xml
<response>
  <foo />
  <result>
    <log>
      <logs count="3">
        <entry id="1">
          <baz>1</baz>
        </entry>
        <entry id="2">
          <baz>2</baz>
        </entry>
        <entry id="3">
          <baz>3</baz>
        </entry>
      </logs>
    </log>
  </result>
  <bar />
</response>
```

I'm trying to create an iterator that traverses through the three `<entry>` elements like this:

``` php
$xml_element_iterator = new \XMLElementIterator($xml_reader);
$iterator = new \XMLElementXpathFilter(
  $xml_element_iterator,
  '/response/result/log/logs/entry'
);
foreach ($iterator as $entry) {
  $simplexml = $entry->getSimpleXMLElement();
  // The following line fails
  echo "entry: ".(string) $simplexml->baz."\n";
}
```

The line fails because `$simplexml` is NULL. Probably because the xpath filter only gets fed an empty `<entry>` element. Is this a bug or is it deliberate and should I use another iterator?

**Comment by hakre** [comment]
Sorry for the late reply, your report just got my attention and I tried to reproduce it following your good description. Unfortunately I was unable to reproduce it. ` $simplexml` never is `null` when I run the scenario you report. Perhaps needles to say, the said failed line works well over here.

This is perhaps an issue with the libxml version. You can obtain it via the according version constants:

~~~
$ php -r 'var_dump(LIBXML_DOTTED_VERSION, PHP_VERSION);'
string(5) "2.9.1"
string(30) "5.6.26-1+deb.sury.org~trusty+1"
~~~

This for example is the configuration I've tested your report against.

**Comment by impcyber** [comment]
`$iterator = new \XMLElementXpathFilter($xml_element_iterator, '//entry');`
Will iterate 3 times, but in structure like this (change node names from <result> to <entry>):
```
<response>
  <foo />
  <entry>
    <log>
      <logs count="3">
        <entry id="1">
          <baz>1</baz>
        </entry>
        <entry id="2">
          <baz>2</baz>
        </entry>
        <entry id="3">
          <baz>3</baz>
        </entry>
      </logs>
    </log>
  </entry>
  <bar />
</response>
```
It iterates only 1 time on parent <entry>, but if we will look here http://www.xpathtester.com/xpath with the same xml-code it return 4 items. 
Is this a bug or standart behaviour?

**Comment by hakre** [comment]
> Is this a bug or standart behaviour?

This was a bug fixed in v0.1.12 @impcyber. Fix credits @jwundrak.
